source/dub/dub.d

```diff
@@ -87,6 +87,37 @@ PackageSupplier[] defaultPackageSuppliers()
 	];
 }
 
+/** Returns a registry package supplier according to protocol.
+
+	Allowed protocols are dub+http(s):// and maven+http(s)://.
+*/
+PackageSupplier getRegistryPackageSupplier(string url)
```

Comment:
I worked around bug #1418 by adapting the string before creating URL.
If this is OK I would leave the method as it is and solve the bug in URL in an independent pr.